### PR TITLE
set default page numbers

### LINKED ISSUE #702: [Images to PDF] Add Set as Default option in Show page numbers
### Steps to reproduce
* Open app 
* Create PDF
* Select Images
* Show page numbers (a dialog will open)

### What to do?
 * Add a checkbox at the bottom: `Set as default`
 * If that checkbox is checked, set the currently selected option as a default option.

### PR DESCRIPTION
**Pull request by jeannewangxin**
# Description

Add a checkbox in the dialog of "Choose page number style",if the checkbox is chosen, it will remember the latest choose and the the next time you open this page you'll find the checkbox and the radiobutton of the latest choice are chosen.If the checkbox "Set as default" is not chosen, your choice will not be preserved.

Fixes #702

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x ] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [x ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -13,6 +13,11 @@ android {
         versionCode 90
         versionName "8.6.3"
     }
+    lintOptions {
```
Comment (vyankatesh24) [inline app/build.gradle]: Revert these changes!

File: app/src/main/res/layout/add_pgnum_dialog.xml [hunk 2]
```diff
@@ -32,4 +32,16 @@
...
+
+    <CheckBox
+        android:id="@+id/set_as_default"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_alignStart="@+id/radioGroup"
+        android:layout_alignParentTop="true"
+        android:layout_marginTop="126dp"
+        android:checked="false"
+        android:text="Set as default"
```
Comment (vyankatesh24) [inline app/src/main/res/layout/add_pgnum_dialog.xml]: Use `Strings.xml`

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 3]
```diff
@@ -866,27 +880,59 @@ void addMargins() {
...
-                    } else if (checkedRadioButtonId == rbOpt2.getId()) {
+                    } else if (checkedRadioButtonId == rbOpt2.getId ()) {
                         mPageNumStyle = Constants.PG_NUM_STYLE_X_OF_N;
-                    } else if (checkedRadioButtonId == rbOpt3.getId()) {
+                    } else if (checkedRadioButtonId == rbOpt3.getId ()) {
                         mPageNumStyle = Constants.PG_NUM_STYLE_X;
                     }
+                    if (cbDefault.isChecked ()) {
+
+                        editor.putString ("pre", mPageNumStyle);
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: Use `Constants.java` to add constants wherever you have used.

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 4]
```diff
@@ -866,27 +880,59 @@ void addMargins() {
         materialDialog.show();
     }
 
+
     private void addPageNumbers() {
+
+        SharedPreferences.Editor editor = pref.edit();
+
+        mPageNumStyle = pref.getString ("pre", null);
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: Use `Constants.java` to add constants wherever you have used.

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 5]
```diff
@@ -142,7 +148,9 @@
     private int mMarginBottom = 38;
     private int mMarginLeft = 50;
     private int mMarginRight = 38;
-    private String mPageNumStyle;
+    private String mPageNumStyle = null;
+    private int mChoseId;
+    SharedPreferences pref;
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: Make private.

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 6]
```diff
@@ -20,14 +21,17 @@
 import android.support.v7.widget.RecyclerView;
 import android.text.Editable;
 import android.text.TextWatcher;
+import android.util.Log;
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: Remove log statements.

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 7]
```diff
@@ -153,6 +161,8 @@ public void onAttach(Context context) {
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
 
+        pref = mActivity.getApplicationContext ().getSharedPreferences("MyPref", MODE_PRIVATE);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: use PreferenceManager.getDefaultSharedPreferences(context) to get sharedpref instance. dont create new typr of prefs for this

File: app/build.gradle [hunk 8]
```diff
@@ -13,6 +13,7 @@ android {
         versionCode 90
         versionName "8.6.3"
     }
+
```
Comment (vyankatesh24) [inline app/build.gradle]: Revert this change!

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 9]
```diff
@@ -153,6 +159,8 @@ public void onAttach(Context context) {
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
 
+        mPref = mActivity.getSharedPreferences("MyPref", MODE_PRIVATE);
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: Change this as suggested by @Swati4star

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 10]
```diff
@@ -279,6 +288,9 @@ void createPdf(boolean isgrayScale) {
                         new CreatePdf(mPdfOptions, mHomePath, ImageToPdfFragment.this).execute();
                     }).onNegative((dialog1, which) -> createPdf(isgrayScale)).show();
                 }
+
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: Revert this change!

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 11]
```diff
@@ -866,27 +878,59 @@ void addMargins() {
...
     }
 
+
     private void addPageNumbers() {
+
+        SharedPreferences.Editor editor = mPref.edit();
+
+        mPageNumStyle = mPref.getString (Constants.PREFSTYLE, null);
+        mChoseId = mPref.getInt (Constants.PREFID, -1);
+
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: Remove blank lines between.

File: app/src/main/res/values-es/strings.xml [hunk 12]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<resources>
+<resources xmlns:tools="http://schemas.android.com/tools">
```
Comment (vyankatesh24) [inline app/src/main/res/values-es/strings.xml]: Revert this change!

File: app/src/main/java/swati4star/createpdf/util/Constants.java [hunk 13]
```diff
@@ -102,4 +102,8 @@
 
     public static final String VERSION_NAME = "VERSION_NAME";
 
+    public static final String PREFSTYLE = "pref";
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/util/Constants.java]: Use some intuitive name related to page numbers.

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 14]
```diff
@@ -142,7 +146,9 @@
     private int mMarginBottom = 38;
     private int mMarginLeft = 50;
     private int mMarginRight = 38;
-    private String mPageNumStyle;
+    private String mPageNumStyle = null;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: there shouldnt be a need to initialize this value to null

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 15]
```diff
@@ -142,7 +146,9 @@
...
+    private int mChoseId;
+    private SharedPreferences mPref;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: change variable name to `mSharedPreferences`

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 16]
```diff
@@ -124,6 +127,7 @@
     @BindView(R.id.tvNoOfImages)
     TextView mNoOfImages;
 
+    public static int choosePageStyle;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: remove this variable

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 17]
```diff
@@ -1,6 +1,7 @@
 package swati4star.createpdf.fragment;
 
 import android.Manifest;
+import android.annotation.SuppressLint;
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: remove this import

File: app/src/main/java/swati4star/createpdf/util/Constants.java [hunk 18]
```diff
@@ -102,4 +102,8 @@
 
     public static final String VERSION_NAME = "VERSION_NAME";
 
+    public static final String PREFSTYLE = "pref_page_style";
+    public static final String PREFID = "id";
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/Constants.java]: this is a very generic name. change it

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 19]
```diff
@@ -153,6 +156,8 @@ public void onAttach(Context context) {
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
 
+        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(mActivity);
+        mPageNumStyle = mSharedPreferences.getString (Constants.PREF_PAGE_STYLE, null);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: move this line to `resetValues` function

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 20]
```diff
@@ -153,6 +156,8 @@ public void onAttach(Context context) {
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
 
+        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(mActivity);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: there is no need to initialize this again. This is already initialized

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 21]
```diff
@@ -264,9 +269,10 @@ void createPdf(boolean isgrayScale) {
             } else {
                 final String filename = input.toString();
                 FileUtils utils = new FileUtils(mActivity);
+
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: remove this extra line added by you

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java (hunk 19 above)
Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: this isnt done yet

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java [hunk 22]
```diff
@@ -791,13 +794,14 @@ private void resetValues() {
                 DEFAULT_PAGE_SIZE));
         mPdfOptions.setPasswordProtected(false);
         mPdfOptions.setWatermarkAdded(false);
+        mSharedPreferences = PreferenceManager.getDefaultSharedPreferences(mActivity);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: why have you added this?

Please do all changes at once. This is taking too much iterations!!!!!

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java (hunk 19 above)
Comment (jeannewangxin) [reply app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: So I should remove the number 160 line to resetValues?

File: app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java (hunk 22 above)
Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: then change it?

Comment (jeannewangxin) [reply app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: I have moved this lines to resetValues function, it is  right?
And do I need any other changes?

Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: There still are changes, but i'll do them & merge PRs.
There have al;ready been so many iterations for a small change

Comment (jeannewangxin) [reply app/src/main/java/swati4star/createpdf/fragment/ImageToPdfFragment.java]: OK, apologize for that, and thanks.

**Comment by Swati4star** [comment]
merged in https://github.com/Swati4star/Images-to-PDF/commit/9a613a3784f2e9b7a8940ef52b78afb4d6f670cc
